spikeinterface/sortingcomponents/waveforms/neural_network_denoiser.py

```diff
@@ -0,0 +1,116 @@
+from pathlib import Path
+import json
+from typing import List, Optional
+
+import torch
+from torch import nn
+from huggingface_hub import hf_hub_download
+
+
+from spikeinterface.core import BaseRecording
+from spikeinterface.sortingcomponents.peak_pipeline import PipelineNode, WaveformExtractorNode
+from .waveform_utils import to_temporal_representation, from_temporal_representation
+
+
+class SingleChannelToyDenoiser(PipelineNode):
+    def __init__(
+        self, recording: BaseRecording, return_output: bool = True, parents: Optional[List[PipelineNode]] = None
+    ):
+        super().__init__(recording, return_output=return_output, parents=parents)
+
+        # Find waveform extractor in the parents
+        try:
+            waveform_extractor = next(parent for parent in self.parents if isinstance(parent, WaveformExtractorNode))
+        except (StopIteration, TypeError):
+            exception_string = f"TemporalPCA should have a {WaveformExtractorNode.__name__} in its parents"
+            raise TypeError(exception_string)
+
+        self.assert_model_and_waveform_temporal_match(waveform_extractor)
+
+        # Load model
+        self.denoiser = self.load_model()
+
+    def assert_model_and_waveform_temporal_match(self, waveform_extractor: WaveformExtractorNode):
+        """
+        Asserts that the model and the waveform extractor have the same temporal parameters
+        """
+        # Extract temporal parameters from the waveform extractor
+        waveforms_ms_before = waveform_extractor.ms_before
+        waveforms_ms_after = waveform_extractor.ms_after
+        waveforms_sampling_frequency = waveform_extractor.recording.get_sampling_frequency()
+
+        # Load the model temporal parameters
+        repo_id = "SpikeInterface/test_repo"
+        filename = "toy_model_peak_interval.json"
```

Comment:
My thinking was that this is used for the specic temporal parameters that determine whether the model matches with the waveform extractor (sampling_frequency, ms_before, ms_after, etc). In opposition to the structural parameters and weights are determined in the other pickled file. A similar story with the TemporalPCA. That is, it is not all the parameters but the temporal matching ones.

From that perspective I think that `params.json` is not specific enough. If want to introduce, for other models that use other set of parameters (not just structural and temporal matching like PCA and the CNN here) we would create an unecessary ambiguity. 

I think that something minimal like `temporal_parameters.json` or something of the sort is better.

Reply:
I disagree here @h-mayorquin 

In this case the params might only have temporal params info, but what if a model is built on some geometry or other parameters? In that case we would extend this params name, e.g. `temporal_geometry_local_radius.json`?

In my opinion it's better to unify all names for models as `params.json`, so each model folder will have a `model.*` (.pt, .pkl) and a `params.json`

Reply:
I see how you view it. All the parameters, not only temporal, should go in that file. 

I was not thinking on this because for the neural network models the network parameters are in the pickle file so we already have two parameters files now. From there, I thought we would have models with 3, 4, and so on so I wanted to be specific. Maybe we won't require more than 2 and your approach makes more sense.

So, let's stick with the unification -all parameters in that file- and split later if it makes sense.

Reply:
Done in the last commit.